### ISSUE TITLE
no-confusing-arrow and no-extra-parens conflict

### ISSUE THREAD

**Issue by LJNeon** [eslint/eslint#10997]
**Tell us about your environment**

* **ESLint Version:** 5.7.0
* **Node Version:** 10.12.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>
(Our actual configuration is hundreds of lines long so I'm only including the relevant rules)

```yml
env: {
  es6: true,
  node: true
}

extends: "eslint:recommended"

rules: {
  no-confusing-arrow: [2, {allowParens: true}],
  no-extra-parens: [2, "all", {nestedBinaryExpressions: false}]
}

parserOptions: {ecmaVersion: 2018}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
array.map(i => format == null ? i.name : format(i));
```
```bash
eslint . --ext .js
```

**What did you expect to happen?**
Once we added parenthesis to solve the `no-confusing-arrow` error, it would lint without errors.

**What actually happened? Please include the actual, raw output from ESLint.**
Before adding the parenthesis:
```
/src/utility/TypeReaderUtil.js
  25:17  error  Arrow function used ambiguously with a conditional expression  no-confusing-arrow
```
After adding the parenthesis:
```
/src/utility/TypeReaderUtil.js
  25:22  error  Unnecessary parentheses around expression  no-extra-parens
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report. You can use the [`enforceForArrowConditionals`](https://eslint.org/docs/rules/no-extra-parens#options) option for the `no-extra-parens` rule to avoid this conflict.

**Comment by LJNeon** [comment]
Thank you.
